src/test/base58_tests.cpp

```diff
@@ -81,19 +85,31 @@ BOOST_AUTO_TEST_CASE(base58_DecodeBase58)
     BOOST_CHECK(!DecodeBase58Check("3vQB7B6MrGQZaxCuFg4oh\0" "0IOl"s, result, 100));
 }
 
-BOOST_AUTO_TEST_CASE(base58_random_encode_decode)
+BOOST_AUTO_TEST_CASE(base58_random_encode_decode_with_optional_spaces)
 {
     for (int n = 0; n < 1000; ++n) {
-        unsigned int len = 1 + InsecureRandBits(8);
-        unsigned int zeroes = InsecureRandBool() ? InsecureRandRange(len + 1) : 0;
+        auto len = 1 + InsecureRandBits(8);
+        auto zeroes = InsecureRandBool() ? InsecureRandRange(len + 1) : 0;
         auto data = Cat(std::vector<unsigned char>(zeroes, '\000'), g_insecure_rand_ctx.randbytes(len - zeroes));
-        auto encoded = EncodeBase58Check(data);
+
+        auto leadingSpaces = InsecureRandBool() ? std::string(InsecureRandRange(10), ' ') : "";
+        auto trailingSpaces = InsecureRandBool() ? std::string(InsecureRandRange(10), ' ') : "";
```

Comment:
nit:  Probably outside the scope of this PR (this PR is adding tests, so business logic changes are extraneous), but at first glance, seems like these statements could be simplified, since `InsecureRandRange()` can return 0, the `std::string` constructor can handle 0 count, and string operator+ can handle empty string addition.  Maybe I'm missing something?

For example:

```diff
diff --git a/src/test/base58_tests.cpp b/src/test/base58_tests.cpp
index 49ef9ff5b5..beb5ef3335 100644
--- a/src/test/base58_tests.cpp
+++ b/src/test/base58_tests.cpp
@@ -92,8 +92,8 @@ BOOST_AUTO_TEST_CASE(base58_random_encode_decode_with_optional_spaces)
         auto zeroes = InsecureRandBool() ? InsecureRandRange(len + 1) : 0;
         auto data = Cat(std::vector<unsigned char>(zeroes, '\000'), g_insecure_rand_ctx.randbytes(len - zeroes));
 
-        auto leadingSpaces = InsecureRandBool() ? std::string(InsecureRandRange(10), ' ') : "";
-        auto trailingSpaces = InsecureRandBool() ? std::string(InsecureRandRange(10), ' ') : "";
+        auto leadingSpaces = std::string(InsecureRandRange(10), ' ');
+        auto trailingSpaces = std::string(InsecureRandRange(10), ' ');
         auto encoded = leadingSpaces + EncodeBase58Check(data) + trailingSpaces;
 
         std::vector<unsigned char> decoded;
```

Reply:
Thanks for checking @tdb3, the results would be similar, but since I assumed the spaces are rare in reality, I gave it different odds.
In my impl the probability that we won't have any leading (or trailing) spaces was `50% + 50%*10%`, in your impl it's 10%, so spaces would be in most samples.
I'm fine with both.

Reply:
Thanks, that's right (higher probability of no spaces over rand range alone). I don't have a preference, just an observation.

Reply:
I like all this, but it would be better to explain in the PR description why it is important/beneficial to include this change since this increases the test complexity. 

Probably would be even better to have a separate test case to check for leading and trailing spaces. This test case is intended to check for leading zeros and now it checks for three concerns (leading zeros, leading spaces, and trailing spaces) instead of one.

Reply:
> it checks for three concerns

That's a property based test, i.e. for random valid inputs it checks that certain conditions hold - in this case a roundtrip, that decoding an encoded results in the original trimmed value.
It's meant to find corner cases that we haven't though of, that's its single concern, we have separate tests for each corner case we know about.

Reply:
Added short explanations to the commit message